tests/sles4sap/sapconf.pm

```diff
@@ -35,6 +35,52 @@ sub check_profile {
       unless ($output =~ /Current active profile: $profile/);
 }
 
+sub run_developers_tests {
+    my $devel_repo = 'https://gitlab.suse.de/AngelaBriel/sapconf-test/repository/master/archive.tar.gz';
+    my $log        = '/tmp/sapconf_test.log';
+
+    # Download and unpack the test scripts supplied by the developers
+    # Record soft failure and continue if it can not be downloaded
+    my $ret    = script_run "curl -k $devel_repo > /tmp/sapconf-test.tar.gz";
+    my $output = script_output "file /tmp/sapconf-test.tar.gz";
+    unless (defined $ret and $ret == 0 and $output =~ /gzip compressed data/) {
+        record_soft_failure 'Could not download developer test script';
+        return;
+    }
+    type_string "cd /tmp\n";
+    assert_script_run "tar -zxf /tmp/sapconf-test.tar.gz";
+
+    # Run script as is and upload results
+    type_string "cd sapconf-test-master-*\n";
```

Comment:
using type_string means "I don't care if it passed or failed". But actually you do care. If this command will fail your next script_output will fail . But for some time you will stay confused why actually it was failing

Reply:
Actually I don't mind if the `cd` command fails, as the next command is just an `script_output 'ls'` and that should always work (if it fails, I have another problem, not specifically related to sapconf). But I guess I do want to have different soft failures for when I don't find the test script and when I don't find the directory in the tar file, so I will re-write this part. Thanks.